src/mediaStreams.ts

```diff
@@ -0,0 +1,395 @@
+/*
+ * @Author: Elias Sun(elias.sun@ringcentral.com)
+ * @Date: Dec. 15, 2018
+ * Copyright © RingCentral. All rights reserved.
+ */
+
+'use strict';
+
+/**
+ * @Supported browsers: @Chrome  @Firefox @Safari
+ *
+ * @Section1 @type MediaStreams public interfaces
+ *
+ * @release : release the resource after the call is ended.
+ * @param : none
+ *
+ * @reconnectMedia : reconnect media streams in a call at any time.
+ * @param : none
+ *
+ * @getMediaStats : get the media RTP statistics
+ * @param1 onMediaStat : @optional  @callback function to receive the RTP statistics report.
+ *   Ways to receive media report:
+ *   @way1 onMediaStat = function(report) {...}
+ *   @way2 session.on("rtpStat") to listen on the event
+ *   @way3 session.onRTPStat = function(report)
+ *   @way4 session.mediaStreams.onRTPStat = function(report)
+ * @param2 interval : @optional  the interval in seconds to fetch a media statistics report. 1 second by default.
+ * @return: @inboundRtpReport : bytesReceived, jitter, packetsLost, packetsReceived, mediaType, fractionLostIn
+ *          @outboundRtpReport : bytesSent, packetsSent, mediaType
+ *          @rttMS : currentRoundTripTime
+ *
+ * @stopMediaStats :stop the media statistics
+ * @param : none
+ *
+ * @Section2 @type MediaStreams public properties
+ *
+ * @property onRTPStat : @optional @callback function to receive the RTP statistics report.
+ * @property onMediaConnectionStateChange : @optional @callback function to receive media connectionState
+ *   Ways to receive the media connection state
+ *   @way1 session.onMediaConnectionStateChange = function(session, event) {...}
+ *   @way2 session.mediaStreams.onMediaConnectionStateChange = function(session, event) {...}
+ *   @way3 session.on(event)  event = element in connectionState
+ *
+ * @Section3 @type MediaStreams public events
+ * @connectionState : media connection state. @emit :
+ *   @state1 'mediaConnectionStateNew' : A new RTCPeerConnection is created.
+ *   @state2 'mediaConnectionStateChecking' : A new RTCPeerConnection is created.
+ *   @state3 'mediaConnectionStateConnected' : RTCPeerConnection media connection is connected.
+ *   @state4 'mediaConnectionStateCompleted' : RTCPeerConnection media connection is ready.
+ *   @state5 'mediaConnectionStateFailed' : RTCPeerConnection media connection is failed.
+ *   @state6 'mediaConnectionStateDisconnected': RTCPeerConnection media connection is disconnected.
+ *   @state7 'mediaConnectionStateClosed' : RTCPeerConnection media connection is closed.
+ */
+
+export default class MediaStreams {
```

Comment:
I can't find how this file is used in the rest of WebPhone... Where's the import?